test/test_images/helloworld-userport/helloworld.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+)
+
+func handler(w http.ResponseWriter, r *http.Request) {
+	log.Print("Hello world received a request.")
+	fmt.Fprintln(w, "Hello World! How about some tasty noodles?")
+}
+
+func main() {
+	flag.Parse()
+	log.Print("Hello world app started.")
+
+	http.HandleFunc("/", handler)
+	http.ListenAndServe(":8888", nil)
```

Comment:
Can you listen on the environment variable port rather than 8888? That would be further validation that the environment variable is completely plumbed through.

Reply:
ok